app/client/components/DetailView.js

```diff
@@ -75,18 +76,37 @@ function DetailView(gristDoc, viewSectionModel) {
 
   //--------------------------------------------------
   // Set up DOM event handling.
+  this.lastElemSelected = null;
 
   // Clicking on a detail field selects that field.
   this.onEvent(this.viewPane, 'mousedown', '.g_record_detail_el', function(elem, event) {
     this.viewSection.hasFocus(true);
     var rowModel = this.recordLayout.getContainingRow(elem, this.viewPane);
     var field = this.recordLayout.getContainingField(elem, this.viewPane);
     commands.allCommands.setCursor.run(rowModel, field);
+    this.lastElemSelected = elem;
   });
 
   // Double-clicking on a field also starts editing the field.
   this.onEvent(this.viewPane, 'dblclick', '.g_record_detail_el', function(elem, event) {
     this.activateEditorAtCursor();
+    this.lastElemSelected = elem;
+  });
+
+  // Clicking on a field also starts editing the field but only on certain cases and if the field are preselected
+  const simpleClickUnacceptedSelectors = [".formula_field", ".widget_checkbox", ".widget_switch"]
+  this.onEvent(this.viewPane, 'click', '.g_record_detail_el', function(elem, event) {
+    if (
+      elem === this.lastFieldSelected
+      && !isNarrowScreen()
+      // Trick to avoid event to be triggered in other context,
+      // and the error `UnexpectedAlertOpenError: unexpected alert open: {Alert text : }`
+      && elem.querySelector(".g_record_detail_el ")
+      && simpleClickUnacceptedSelectors.reduce((formula, current) => formula && !elem.querySelector(current), true)
+    ) {
+      this.activateEditorAtCursor();
+    }
+    this.lastFieldSelected = elem;
```

Comment:
I don't think it a good idea because if we have 2 cards on the same page the `cursor.fieldIndex()` has same value for two different fields because the index is relative to the widget

Reply:
That's correct. The section of the field and cursor would also need to be checked, but I thought of another approach that might be simpler. Suppose `Cursor.ts` had the following instance variable:

```
this.previousPosition = ko.observable(optCursorPos);
```

Then the condition in `DetailView.js` could be changed to:

```
this.cursor.currentPosition().fieldIndex === this.cursor.previousPosition().fieldIndex
```

Tracking the previous cursor position may require some care, but should essentially boil down to updating the 
`previousPosition` observable in `Cursor.setCursorPos` with a snapshot of `currentPosition`, before changes are made to it.

Reply:
Just compare field id works fine too, it an easy and reliable solution, what do you think ? (it's what i've done)